packages/react/src/ReactHooks.js

```diff
@@ -218,24 +218,26 @@ export function useSyncExternalStore<T>(
 
 export function useCacheRefresh(): <T>(?() => T, ?T) => void {
   const dispatcher = resolveDispatcher();
-  // $FlowFixMe This is unstable, thus optional
+  // $FlowFixMe[not-a-function] This is unstable, thus optional
   return dispatcher.useCacheRefresh();
 }
 
 export function use<T>(usable: Usable<T>): T {
   const dispatcher = resolveDispatcher();
-  // $FlowFixMe This is unstable, thus optional
+  // $FlowFixMe[not-a-function] This is unstable, thus optional
   return dispatcher.use(usable);
 }
 
 export function useMemoCache(size: number): Array<any> {
   const dispatcher = resolveDispatcher();
-  // $FlowFixMe This is unstable, thus optional
+  // $FlowFixMe[not-a-function] This is unstable, thus optional
   return dispatcher.useMemoCache(size);
 }
 
-export function useEffectEvent<T>(callback: T): void {
+export function useEffectEvent<Args, Return, F: (...Array<Args>) => Return>(
```

Comment:
This is a little bit shorter, we don't have to define the return type in the type constraint for F:
```suggestion
export function useEffectEvent<Args, Return, F: (...Array<Args>) => Return>(
export function useEffectEvent<Args, F: (...Args) => mixed>(
```

Reply:
Applied in [facebook/react@`9cf21fb` (#26468)](https://github.com/facebook/react/pull/26468/commits/9cf21fb9f227cedbf6c0dc896fcf32e856d2e296) to make it consistent with the internal types.